lib/natalie/compiler2/instructions/const_find_instruction.rb

```diff
@@ -14,11 +14,14 @@ def to_s
       end
 
       def generate(transform)
-        transform.push("self->const_find(env, #{name.inspect}_s, Object::ConstLookupSearchMode::NotStrict)")
+        namespace = transform.pop
+        transform.push("#{namespace}->const_find(env, #{name.inspect}_s, Object::ConstLookupSearchMode::NotStrict)")
       end
 
       def execute(vm)
-        vm.push Object.const_get(@name)
+        namespace = vm.pop
+        namespace = namespace.class unless namespace.respond_to?(:const_get)
```

Comment:
The reason I did it this way was because MRI's `main` does *not* have a `const_get`...

```
→ irb
irb(main):001:0> FOO = 1
=> 1
irb(main):002:0> const_get(:FOO)
(irb):2:in `<main>': undefined method `const_get' for main:Object (NoMethodError)
Did you mean?  context
	from /Users/tim/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /Users/tim/.rbenv/versions/3.0.3/bin/irb:23:in `load'
	from /Users/tim/.rbenv/versions/3.0.3/bin/irb:23:in `<main>'
irb(main):003:0> self.const_get(:FOO)
(irb):3:in `<main>': undefined method `const_get' for main:Object (NoMethodError)
Did you mean?  context
	from /Users/tim/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /Users/tim/.rbenv/versions/3.0.3/bin/irb:23:in `load'
	from /Users/tim/.rbenv/versions/3.0.3/bin/irb:23:in `<main>'
```

Reply:
@FnControlOption any thoughts on this...?

Reply:
Oh, I worked around this by checking if `namespace` is `MainObject` (ee8dae76e19e348df239781ab913980441c177a7)

Reply:
OK, I'm confused about your review here... and that commit... that commit isn't in this PR (maybe you or I rebased it out?) and I'm not sure if you're saying that code is better than this one liner... Can you use more words?

Reply:
I'm just going to go with what is here. If it's wrong, we can correct it in another PR....